File: Backend/login-microsoft365/src/main/java/com/miapp/model/dto/RecursoDigitalDTO.java
package com.miapp.model.dto;

import lombok.Data;
import java.time.LocalDateTime;

@Data
public class RecursoDigitalDTO {
    private Long id;
    private String autor;
    private Long tipoId;
    private String tipoDescripcion;
    private String titulo;
    private String descripcion;
    private String enlace;
    private Integer estado;
    private String usuarioCreacion;
    private String usuarioModificacion;
    private LocalDateTime fechaCreacion;
    private LocalDateTime fechaModificacion;
    private String imagenUrl; // base64 o ruta
    private Long clicks;
}

File: Frontend/sakai-ng-master/src/app/biblioteca/interfaces/RecursoDigitalDTO.ts
export interface RecursoDigitalDTO {
 id?: number;
   autor: string;
   titulo: string;
   tipoId?: number;
   descripcion: string;
   tipoDescripcion?: string;
   enlace: string;
   estado: number;               // 1 = activo, 0 = inactivo
   usuarioCreacion?: string;
   usuarioModificacion?: string;
   fechaCreacion?: string;       // ISO string
   fechaModificacion?: string;   // ISO string
   imagen?: string;              // base64 o URL, se mapea directamente del campo 'imagen'
   imagenUrl?: string;           // URL pública entregada por el backend
   clicks?: number;
}
